.github/workflows/documentation-build.yml

```diff
@@ -6,32 +6,57 @@ on:
       - '*'   # Only re-generate docs when a new tagged version is pushed
   pull_request:
     paths:
-      - '.github/workflows/documentation-build.yml'
       - 'Doxyfile'
-      - 'doxygen_style.css'
+      - 'DoxygenLayout.xml'
+      - '**.h'
+      - '/doc/doxygen/**'
+      - '/doc/doxygen-extra-pages/**'
+      - '/doc/doxygen-groups/**'
+      - '.github/workflows/documentation-*.yml'
   workflow_dispatch:
 
+env:
+  COCKATRICE_REF: ${{ github.ref_name }}    # Tag name if the commit is tagged, otherwise branch name
+
 jobs:
   docs:
     name: Doxygen
     runs-on: ubuntu-latest
 
     steps:
-      - name: Checkout code
+      - name: Checkout
         uses: actions/checkout@v5
 
       - name: Install Doxygen and Graphviz
+        shell: bash
         run: sudo apt-get install -y doxygen graphviz
 
       - name: Generate Documentation
+        id: build_docs
+        shell: bash
         run: doxygen Doxyfile
 
-      - name: Deploy to cockatrice.github.io
-        if: github.event_name != 'pull_request'
+      - name: Upload Documentation
+        id: upload_docs
+        if: steps.build_docs.outcome == 'success'
```

Comment:
do we need to add this check to every step? isn't fail fast the default?

Reply:
but it'd immediately fail the entire thing and not deploy right?

Reply:
Had to add an if condition for the very last one though that it does not trigger if deployment is skipped on PR runs.